openedx/frontend-platform
# feat: MFE configuration at runtime

**Description:**

This draft PR serves as a discussion holder for the work of making mfes capable of being configured during runtime instead of during build.

This PR creates a new function to allow runtime configuration through an API call to the platform. 

 **How to test**
- To allow runtime configuration set a new environment variable MFE_CONFIG_API_URL in the env file and add the api url ( To test this you can use the API from this [PR](https://github.com/openedx/edx-platform/pull/30473) or use an external tool to mock the API response).
- The API should respond with a JSON with the config values: 
```json
{
  "SITE_NAME": "Test Site",
  "LOGO_URL": "https://testimage.com/logo.svg",
  "LOGO_TRADEMARK_URL": "https://testimage.com/logo.svg",
  "LOGO_WHITE_URL": "https://testimage.com/logo.svg",
  "FAVICON_URL": "https://testimage.com/favicon.ico",
  "CSRF_TOKEN_API_PATH": "/csrf/api/v1/token",
  "LANGUAGE_PREFERENCE_COOKIE_NAME": "openedx-language-preference",
  "LOGIN_URL": "http://testsite.example.com/login",
  "LOGOUT_URL": "http://testsite.example.com/logout",
   "STUDIO_BASE_URL": "http://studio.example.com",
  "MARKETING_SITE_BASE_URL": "http://testsite.example.com",
  "ORDER_HISTORY_URL": "http://testsite.example.com/orders",
  "REFRESH_ACCESS_TOKEN_ENDPOINT": "http://testsite.example.com/login_refresh",
}
```

- The initialize process should work normally.
_Note: You can combine buildtime and runtime configuration_

**Note: Related information [FWG Issue](https://github.com/openedx/frontend-wg/issues/103)**

**Merge checklist:**
  - [ ]  Consider running your code modifications in the included example app within frontend-platform. This can be done by running npm start and opening http://localhost:8080/. To test this you can use the API from this [PR](https://github.com/openedx/edx-platform/pull/30473) or use an external tool to mock the API response.
  - [ ]  Consider testing your code modifications in another local micro-frontend using local aliases configured via [the module.config.js file in frontend-build](https://github.com/edx/frontend-build#local-module-configuration-for-webpack). To test this you can use the API from this [PR](https://github.com/openedx/edx-platform/pull/30473) or use an external tool to mock the API response.
  - [ ]  Gather some consensus on the approach

**Post merge:**

- [ ] After the build finishes for the merged commit, verify the new release has been pushed to [NPM](https://www.npmjs.com/package/@edx/frontend-platform).
